### PR TITLE
[HyperElastic] Add draw method to StandardTetrahedralFEMForceField

### PR DESCRIPTION
**Pull request by alxbilger**
It is made common with TetrahedronHyperelasticityFEMForceField

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Component/SolidMechanics/FEM/HyperElastic/src/sofa/component/solidmechanics/fem/hyperelastic/TetrahedronHyperelasticityFEMDrawing.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        points[3].push_back(pa);
+        points[3].push_back(pb);
+    }
+
+    sofa::type::RGBAColor color1;
+    sofa::type::RGBAColor color2;
+    sofa::type::RGBAColor color3;
+    sofa::type::RGBAColor color4;
+
+    if (materialName=="ArrudaBoyce") {
```
Comment (fredroy) [inline Component/SolidMechanics/FEM/HyperElastic/src/sofa/component/solidmechanics/fem/hyperelastic/TetrahedronHyperelasticityFEMDrawing.h:77]: maybe the relation "Material" => color could be in a std:map (or any container "str"-> colors) at compile time ?

Comment (alxbilger) [reply Component/SolidMechanics/FEM/HyperElastic/src/sofa/component/solidmechanics/fem/hyperelastic/TetrahedronHyperelasticityFEMDrawing.h:77]: I agree and I thought about it but, you know, copy/paste was quicker

**Comment by damienmarchal** [comment]
Hi @alxbilger 

Thanks for the PR.

This reminds me an open discussion about where to put drawing if this is supposed to be a common feature.

Normally in SOFA there is two draw method one called draw(), and the other is drawVisual
- use the draw() method only for cheap debugging/utilitary. 
- in-herit from visual model and implement a real drawVisual() things if this is part of the behavior you expect. 

So shoulnd't such kind of code be moved into a TetrahedronFieldDrawing component taking a link to tetrahedron mesh and a scalar field to interpolate ? This would make it much more generic and reusable.

**Comment by alxbilger** [comment]
@damienmarchal thanks for your suggestion. It makes sense.
I am not against making things more generic, but here I just noticed that StandardTetrahedralFEMForceField does not show up when I wanted do display force fields (similar to the other tetra force fields). To make a quick fix, I just used the same code than TetrahedronHyperelasticityFEMForceField.
If it is decided to make a deeper refactoring, I'll quit this job as it is not my primary goal.

**Comment by damienmarchal** [comment]
@alxbilger sure. 

Actually I'm looking at Subsettopology for your PR (#2825) and my eyes saw its draw function. Don't :) 
When I saw it, I think we have so much duplicated code in every components and that this propagate to the Data<d_showBlablab>.  But doing such work is clearly something to do small step by step during our dev session or for a coding sprint during STC (@hugtalbot) with pizza's.

**Comment by damienmarchal** [comment]
For future PR: 
- add a "Visualization" panel with a showObject data to activate/deactive the rendering.
- add a component inhering from VisualModel displaying des champs de champs scalaire/vectoriel sur un support d'interpolation.

**Pull request merged**